src/mongo/shell.ts

```diff
@@ -98,6 +100,16 @@ export class Shell {
 				5000);
 			const disposable = this.onResult.event(result => {
 				disposable.dispose();
+
+				if (result.code) {
```

Comment:
What is `result`? Could it ever be undefined? And do we know `code` will always indicate an error or could it be a successful code?

Reply:
I ran a sample command. It looks like there is no code if the execution is hitherto successful. I might have to look at a more formal documentation. 

These were the results on each trigger of the event. 
- "switched to db trial00" 
- "{ "acknowledged" : true, "matchedCount" : 0, "modifiedCount" : 0 }" 

That said, I'd prefer a `if(result && result.code)` just in case.